mllib/src/main/scala/org/apache/spark/mllib/feature/HashingTF.scala

```diff
@@ -97,3 +126,41 @@ class HashingTF(val numFeatures: Int) extends Serializable {
     dataset.rdd.map(this.transform).toJavaRDD()
   }
 }
+
+object HashingTF {
+
+  private[spark] val Native: String = "native"
+
+  private[spark] val Murmur3: String = "murmur3"
+
+  private[spark] val supportedHashAlgorithms = Set(Native, Murmur3)
```

Comment:
Can you store this as an Array in case someone makes this public later on (and does not realize it needs to be Java-friendly)?